helpers/press-release-sparql-queries.js

```diff
@@ -0,0 +1,140 @@
+		PREFIX mu: ${sparqlEscapeUri('http://mu.semte.ch/vocabularies/core/')}
+		PREFIX ebucore: ${sparqlEscapeUri('http://www.ebu.ch/metadata/ontologies/ebucore/ebucore#')}
+		PREFIX ext: ${sparqlEscapeUri('http://mu.semte.ch/vocabularies/ext/')}
+		PREFIX vlpt: ${sparqlEscapeUri('http://themis.vlaanderen.be/id/publicatie-taak/')}
+		PREFIX adms: ${sparqlEscapeUri('http://www.w3.org/ns/adms#')}
+		PREFIX dct: ${sparqlEscapeUri('http://purl.org/dc/terms/')}
+		PREFIX prov: ${sparqlEscapeUri('http://www.w3.org/ns/prov#')}
+		`;
+
+export async function findPressReleasesWithPublicationEvents(id) {
+
+	try {
+		const queryResult = await query(`
+			 ${PREFIXES}
+			 
+			 SELECT ?graph ?pressRelease ?publicationEvent ?publicationStartDateTime ?started ?publicationChannels
+			 WHERE {
+				GRAPH ?graph {
+					?pressRelease 					mu:uuid 									${sparqlEscapeString(id)} ;
+									       			ebucore:isScheduledOn 						?publicationEvent .
+					OPTIONAL { ?publicationEvent 	ebucore:publicationStartDateTime			?started }
+					OPTIONAL { ?publicationEvent 	ebucore:publishedStartDateTime	 			?publicationStartDateTime }
+				}
+			 }
+			 LIMIT 1
+			 `);
+
+		if (!queryResult.results.bindings.length) {
+			return null;
+		} else {
+			const bindings = queryResult.results.bindings[0];
+
+			return {
+				bindings,
+				pressRelease: bindings.pressRelease.value,
+				graph: bindings.graph.value,
+				publicationEvent: bindings.publicationEvent.value,
+				plannedStartDate: bindings.publicationStartDateTime ? bindings.publicationStartDateTime.value : undefined,
+				started: bindings.started ? bindings.started.value : undefined,
+			};
+		}
+	} catch (err) {
+		throw err;
+	}
+
+}
+
+export function removeFuturePublicationDate(graph, pressRelease) {
+	return query(`
+		${PREFIXES}
+		
+		DELETE {
+			GRAPH ${sparqlEscapeUri(graph)} {
+				?publicationEvent	ebucore:publishedStartDateTime	?publishedStartDateTime .
+			}
+		} WHERE {
+			GRAPH ${sparqlEscapeUri(graph)} {
+				${sparqlEscapeUri(pressRelease)}		ebucore:isScheduledOn 			?publicationEvent .
+				?publicationEvent						ebucore:publishedStartDateTime 	?publishedStartDateTime .
+			}
+		}
+	`);
+}
+
+export function setPublicationStartDateTimeAndPublishedStartDateTime(graph, pressRelease, dateTime) {
+	return query(`
+		${PREFIXES}
+		
+		INSERT {
+			GRAPH ${sparqlEscapeUri(graph)} {
+				?publicationEvent	ebucore:publicationStartDateTime 	${sparqlEscapeDateTime(dateTime)} ;
+									ebucore:publishedStartDateTime 		${sparqlEscapeDateTime(dateTime)} .
+			}
+		} WHERE {
+			GRAPH ${sparqlEscapeUri(graph)} {
+				${sparqlEscapeUri(pressRelease)}	ebucore:isScheduledOn 				?publicationEvent .
+			}
+		}
+	`);
+}
+
+export function getPublicationChannelsByPressRelease(graph, pressRelease) {
+	return query(`
+		${PREFIXES}
+		
+		SELECT ?pressRelease ?pubChannel
+		WHERE {
+			GRAPH ${sparqlEscapeUri(graph)} {
+				${sparqlEscapeUri(pressRelease)}		ext:publicationChannels 	?pubChannel .
```

Comment:
Bij de eerste review over het hoofd gezien. Hier moet je de publication-channels nemen die gelinkt zijn aan het publication-event (dat op zijn beurt gelinkt is aan het persbericht)